win32ss/user/user32/windows/cursoricon.c

```diff
@@ -1148,6 +1163,21 @@ BITMAP_LoadImageW(
         goto end;
     CopyMemory(pbmiCopy, pbmi, iBMISize);
 
+    TRACE("Size Image %d, Size Header %d, ResSize %d\n",
+        pbmiCopy->bmiHeader.biSizeImage, pbmiCopy->bmiHeader.biSize, ResSize);
+
+    /* Test if this is a GCC windres.exe compiled 16 or 32 bpp bitmap using
+     * BI_BITFIELDS and if so, then a mistake causes it not to include
+     * the bytes for the bitfields. So, we have to substract out the
+     * size of the bitfields previously included from bitmap_info_size.*/
+    if (compr == BI_BITFIELDS && (bpp == 16 || bpp == 32) &&
+        pbmiCopy->bmiHeader.biSizeImage + pbmiCopy->bmiHeader.biSize == ResSize)
```

Comment:
Note that ResSize could stay equal to zero here, in case the `if(fuLoad & LR_LOADFROMFILE)` block is taken instead (see above). Is this OK still?

Reply:
@HBelusca, Honestly, I do not know the answer to your question. I have not seen any Jira tickets that might indicate this problem either. For now, I think that we can proceed with this PR as it is and I will investigate further. I intend to add some debugging DPRINT1's and run a regression test locally to see if I can find a test that triggers this code path and then if found, I will investigate it and create another PR as needed. Thanks.

Reply:
Update:
@HBelusca, I have now researched the code for your question and have a definitive answer for you. For the case that you mention, we are loading the data from a file and in this case we have a BITMAPFILEHEADER at the file's beginning as is shown here:

https://learn.microsoft.com/en-us/windows/win32/api/wingdi/ns-wingdi-bitmapfileheader

Therefore we can use the ```bfOffBits``` member to find the beginning of the image bits.
For resources, this information has been stripped from the source and therefore this is the reason for testing the resource size to find if the BI_BITFIELDS optional data has been included by the resource compiler.
This is always included by Microsoft's rc.exe if the bpp is 16 or 32 and we have BI_BITFIELDS compression, but is not included by the Binutis' windres.exe program in these cases which is the reason for the added testing.